bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.juicenet.internal.api;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDevice;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDeviceStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiInfo;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiTouSchedule;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link JuiceNetApi} is responsible for implementing the api interface to the JuiceNet cloud server
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetApi {
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetApi.class);
+
+    private static final String API_HOST = "https://jbv1-api.emotorwerks.com/";
+    private static final String API_ACCOUNT = API_HOST + "box_pin";
+    private static final String API_DEVICE = API_HOST + "box_api_secure";
+
+    protected String apiToken = "";
+    protected JuiceNetHttp httpApi;
+    @Nullable
+    protected ThingUID bridgeUID;
+
+    public JuiceNetApi(HttpClient httpClient) {
+        this.httpApi = new JuiceNetHttp(httpClient);
+    }
+
+    public void initialize(String apiToken, ThingUID bridgeUID) {
+        this.apiToken = apiToken;
+        this.bridgeUID = bridgeUID;
+    }
+
+    public List<JuiceNetApiDevice> queryDeviceList()
+            throws JuiceNetApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        Map<String, Object> params = new HashMap<>();
+        ContentResponse response;
+
+        params.put("cmd", "get_account_units");
+        params.put("device_id", bridgeUID.getAsString());
+        params.put("account_token", apiToken);
+
+        response = httpApi.httpPost(API_ACCOUNT, params);
```

Comment:
After switching to Jetty, this seems overly complicated and could be simplified into something like:
```suggestion
        Request request = httpClient.newRequest(API_ACCOUNT);
        request.param("cmd", "get_account_units");
        request.param("device_id", bridgeUID.getAsString());
        request.param("account_token", apiToken);
        ContentResponse response = request.send();
```

which would eliminate the need for `JuiceNetHttp`. This would also give you more control for extending with custom headers, etc.

Reply:
request.param is only for parameters on the URI.  This API requires the "parameters" to be sent as json as part of the body.  I'll still attempt to simplify, but likely still need some Map and then us GSON to create the json body.

Reply:
Sorry for misleading, I had mistaken `paramsToBody` for `paramsToUrl` where my comment would apply to the latter, i.e. the `httpGet` variant.